### ISSUE TITLE
Using vendored feature on windows

### ISSUE THREAD

**Issue by mikart143** [cetra3/tmq#17]
Hello!
During building the git version of tmq I occured an error:
```
   Compiling tmq v0.2.1 (https://github.com/cetra3/tmq#455d9254)
error[E0432]: unresolved import `mio::unix`
 --> C:\Users\mikar\.cargo\git\checkouts\tmq-34da698a4e28e3cb\455d925\src\socket.rs:3:11
  |
3 | use mio::{unix::EventedFd, Evented, Poll, PollOpt, Ready, Token};
  |           ^^^^ could not find `unix` in `mio`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `tmq`.

```

My platfrom is Windows with msvc toolchain

**Comment by cetra3** [comment]
Yeah at the moment support for windows is not great.  We're basically using the `epoll` system to register events to listen to which isn't on windows.  There is talk of using wepoll in some way to assist with this.  I'm not sure how widespread support is.

The main problem is the windows API is fairly different for registering events.  I'd love it if someone could take the reigns and implement it, but it would basically need to be a windows specific write for [socket.rs](https://github.com/cetra3/tmq/blob/master/src/socket.rs)

**Comment by mikart143** [comment]
Thanks for the info !

**Comment by bicarlsen** [comment]
Was thinking of giving this a go. Would `tokio`'s [`NamedPipes`](https://docs.rs/tokio/1.32.0/tokio/net/windows/named_pipe/index.html) be a valid solution here?

**Comment by bicarlsen** [comment]
Gave this a [first attempt](https://github.com/bicarlsen/tmq/tree/feat/windows_compatibility). Would love some feedback if anybody has some. Some of the tests fail, mostly with access denied style errors.
